common/include/fs/File.h

```diff
@@ -10,10 +10,13 @@ class Dentry;
 #define O_WRONLY    0x0001
 #define O_RDWR      0x0002
 #define O_CREAT     0x0004
```

Comment:
O_RDONLY, O_WRONLY, O_RDWR and O_CREAT should be shifted as well so that O_RDONLY != 0 to allow check via `(flags & O_RDONLY)`

(hardcoded constants in exe2minixfs need to be adjusted when O_RDWR or O_CREAT are modified)

Reply:
I apparently forgot that for a reason. I changed it locally now (all #defines and the hardcoded flags to `4 | 8`), grub now tells me that the kernel has not been loaded. Going to look into that....

Reply:
It seems as I may not touch O_RDWR and O_CREAT, otherwise exe2minixfs will fail to convert the kernel image. 

A simple fix would be to not touch O_RDWR and O_CREAT (since exe2minixfs needs to be debugged then) and simply rearrange the defines...

Reply:
`int32 fd = current_sb->createFd(current_inode, flag & 0xFFFFFFFB);` in VfsSyscall.cpp drops the O_RDWR flag, causing writes to fail

Reply:
The permission checks in MinixFSFile/RamFSFile read/write should be changed 
from
`if (((flag_ == O_WRONLY) || (flag_ == O_RDWR)) && (mode_ & A_WRITABLE))`
to
`if (((flag_ & O_WRONLY) || (flag_ & O_RDWR)) && (mode_ & A_WRITABLE))`

Reply:
I just saw and adapted that too, thanks for your help!